master/internal/resourcemanagers/provisioner/provisioner.go

```diff
@@ -110,6 +116,13 @@ func (p *Provisioner) provision(ctx *actor.Context) {
 	if p.scaleDecider.updateInstanceSnapshot(instances) {
 		ctx.Log().Infof("found state changes in %d instances: %s",
 			len(instances), fmtInstances(instances))
+
+		if p.scaleDecider.db != nil {
+			err = p.scaleDecider.recordInstanceStats(p.SlotsPerInstance())
```

Comment:
looking at `calculateInstanceStates`, that's where `scaleDecider.stopped` and `disconnected` are filled in. maybe this update should be *after* line 128? otherwise it may be a "tick" late?..

Reply:
done